### PR TITLE
[Page] Align back action button icon with title on small screens

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-internal/issues/1358

### WHAT is this pull request doing?

Slightly shifts the back action button left so that the icon is visually aligned with the page title below our MD breakpoint. 

| Before | After |
| -- | -- | 
|admin web page-back-action laura-griffee us spin dev_store_shop1_products_1(iPhone SE)|admin web page-back-action laura-griffee us spin dev_store_shop1_products_1(iPhone SE) (2)|

| Before (selected) | After (selected)  |
| -- | -- | 
|admin web page-back-action laura-griffee us spin dev_store_shop1_products_1(iPhone SE) (1)|admin web page-back-action laura-griffee us spin dev_store_shop1_products_1(iPhone SE) (3)|

### How to 🎩

- 🌀 Spin: https://admin.web.page-back-action.laura-griffee.us.spin.dev/store/shop1/products/1
- 📕 Storybook: https://5d559397bae39100201eedc1-pzilrdtxuv.chromatic.com/?path=/story/all-components-page--default

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [x] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by bernardojoaogarcia** [comment]
I’m not sure if the best solution here is to just align the back button icon with the page title. Once that back button is hovered over or pressed, it’ll make the spacing margin unsymmetrical compared to the 'more actions' on the right of the page

**Comment by bernardojoaogarcia** [comment]
However, that might be a worthwhile compromise since it's a different type of button, and the back button doesn't have a `fill` in its default state.

**Comment by bernardojoaogarcia** [comment]
CC: @heyjoethomas for mobile alignment

**Comment by heyjoethomas** [comment]
I agree @bernardojoaogarcia Bernardo, I think it is gonna be strange either way because you see the layout shift when you press or hover.

I actually wonder if we could just hide this element all together on mobile. It is already hidden when in a native app and when in mobile safari/chrome you have the browser back button + swipe to go back. Both of these are much more native actions on mobile. That would likely require us to reconfigure the page actions as well. I will dig into this a bit.

**Comment by lgriffee** [comment]
Going to convert this issue back into a draft while possible options are explored!

File: .changeset/hot-news-mix.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+[`Page`] Aligned `Page` back action button icon with title on small screens
```
Comment (chloerice) [inline .changeset/hot-news-mix.md:5]:
```suggestion
Fixed `Page` `backAction` icon misalignment with `title` on small screens
```

**Review by chloerice — DISMISSED** [review]
🎉

File: .changeset/hot-news-mix.md (hunk 1 above)
Comment (zhongzi-rail0039) [reply .changeset/hot-news-mix.md:5]: Going to convert this issue back into a draft while possible options are explored!

**Comment by BaggioGiacomo** [comment]
Hi there!

Shouldn't the `SkeletonPage` have the same behaviour? 

This is the skeleton page I show the merchant while fetching the data used by the page:
image

I expect the title to be on the green square, am I wrong?
This is the page the merchant sees after the loading page:
image

In addition to causing CLS, isn't this bad to see?
If the title is ok to be shown under the back button, can you also make the skeleton page to be the same?

**Pull request closed without merging**